src/ci/run.sh

```diff
@@ -183,6 +183,9 @@ else
     RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set llvm.static-libstdcpp"
   fi
 
+  # Download GCC from CI on test builders
+  RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set gcc.download-ci-gcc=true"
```

Comment:
Note that for GCC, the download logic is simplified, as I'm not sure if we need `if-unchanged`. `true` means: "if there are changes, build, if there are no changes, download".